Billing.API/Services/Invoice/DummyInvoiceProviderService.cs

```diff
@@ -7,23 +7,32 @@ namespace Billing.API.Services.Invoice
 {
     public class DummyInvoiceProviderService : IInvoiceService
     {
-        public async Task<IEnumerable<InvoiceListItem>> GetInvoices(string clientId)
+        public async Task<IEnumerable<InvoiceListItem>> GetInvoices(string origin, int clientId)
         {
-            if (clientId.In("000000000000000", "999999999999999"))
-                return await Task.Run(() => new List<InvoiceListItem>());
+            if (clientId <= 0)
+                return null!;
+
+            string clientPrefix;
+
+            if (origin.EqualsIgnoreCase("doppler"))
+                clientPrefix = "CD";
+            else if (origin.EqualsIgnoreCase("relay"))
+                clientPrefix = "CR";
+            else
+                return null!;
 
             return await Task.Run(() => new List<InvoiceListItem>()
```

Comment:
Please, do not run a task when the result is available

```suggestion
            return await Task.FromResult(new List<InvoiceListItem>()
```